hr_timesheet_hours_billed/models/account_analytic_line.py

```diff
@@ -0,0 +1,39 @@
+# Copyright 2023-nowdays Cetmix OU (https://cetmix.com)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl)
+from odoo import api, fields, models
+
+
+class AccountAnalyticLine(models.Model):
+    _inherit = "account.analytic.line"
+
+    approved = fields.Boolean(default=False, inverse="_inverse_approved")
+    approved_user_id = fields.Many2one(
+        "res.users",
+        string="Approved by",
+        index=True,
+        tracking=True,
+        readonly=True,
+    )
+    approved_date = fields.Datetime(copy=False, readonly=True)
+    unit_amount_billed = fields.Float(string="Hours Billed")
+
+    # set approved_user_id and date if approved state change
+    def _inverse_approved(self):
+        user = self.env.user
+        now = fields.Datetime.now()
+        for record in self.filtered(lambda rec: rec.approved):
```

Comment:
Rule of thumb: if you don't need a variable w/ the result of `filtered` to play with later, avoid this approach because you loop twice (possibly on the same recordset):
```suggestion
        for record in self:
            if not rec.approved:
                continue
```